askama_shared/src/generator.rs

```diff
@@ -804,26 +804,34 @@ impl<'a, S: std::hash::BuildHasher> Generator<'a, S> {
         buf.writeln(";")
     }
 
-    fn is_shadowing_variable(&self, var: &Target<'a>) -> bool {
+    fn is_shadowing_variable(&self, var: &Target<'a>) -> Result<bool, CompileError> {
         match var {
             Target::Name(name) => {
                 let name = normalize_identifier(name);
                 match self.locals.get(&name) {
                     // declares a new variable
-                    None => false,
+                    None => Ok(false),
                     // an initialized variable gets shadowed
-                    Some(meta) if meta.initialized => true,
+                    Some(meta) if meta.initialized => Ok(true),
                     // initializes a variable that was introduced in a LetDecl before
-                    _ => false,
+                    _ => Ok(false),
                 }
             }
             Target::Tuple(_, targets) => targets
                 .iter()
-                .any(|target| self.is_shadowing_variable(target)),
+                .find_map(|target| match self.is_shadowing_variable(target) {
+                    Ok(false) => None,
+                    outcome => Some(outcome),
+                })
+                .unwrap_or(Ok(false)),
```

Comment:
Transforming from `Ok(false)` to `None`, then back to `Ok(false)` doesn't seem like a great way to do this. At this point it might be better to just use a `for`-loop?

Reply:
Yeah, a loop looks better. Changed.